src/renderers/shared/fiber/__tests__/ReactIncrementalSideEffects-test.js

```diff
@@ -1129,6 +1133,14 @@ describe('ReactIncrementalSideEffects', () => {
       null,
     ]);
 
+    expectDev(normalizeCodeLocInfo(console.error.calls.argsFor(0)[0])).toBe(
+      'Warning: Stateless function components cannot be given refs. ' +
+      'Attempts to access this ref will fail. Check the render method ' +
+      'of `FunctionalComponent`.\n' +
```

Comment:
Thank you for the thorough review, @gaearon. This is my own remaining question. While this test passes this looks wrong to me. (The same set of assertions fails in ReactStatelessComponent with fiber on). 

I would expect the `ownerName` to be `Foo` rather than `FunctionalComponent`. If I cahnge the `getCurrentFiberOwnerName()` to return the `fiber._debugOwner` for `FunctionalComponent`s then I get `Foo`...